packages/core/src/shared/utilities/cacheUtils.ts

```diff
@@ -160,15 +154,34 @@ export function createDiskCache<V, K>(
                     return log('file not found', key)
                 }
 
-                throw ToolkitError.chain(error, `Failed to delete "${target}"`, {
-                    code: 'FSDeleteFailed',
-                    details: { key },
-                })
+                throw createDiskCacheError(error, 'CLEAR', target, key)
             }
 
             log(`deleted (reason: ${reason})`, key)
         },
     }
+
+    /** Helper to make a disk cache error */
+    function createDiskCacheError(error: unknown, operation: 'LOAD' | 'SAVE' | 'CLEAR', target: string, key: K) {
+        return DiskCacheError.chain(error, `${operation} failed for '${target}'`, {
+            details: { key },
+        })
+    }
+}
+
+/**
+ * Represents a generalized error that happened during a disk cache operation.
+ *
+ * For example, when SSO refreshes a token a disk cache error can occur when it
+ * attempts to read/write the disk cache. These errors can be recoverable and do not
```

Comment:
> These errors can be recoverable
(newbie questions): How will we identify which errors are recoverable? Or will that happen somewhere else?

Reply:
In the context of a disk cache error during token refresh, when the issue is related to the filesystem it just means
that if the disk cache error is fixed then we should be able to refresh the token.

So technically all disk related errors should be recoverable since the problems we saw were related to permissions, storage space, file was busy, ... and the current hope is the user either sees the message and uses intuition to fix it, or reports it to us